jax/interpreters/pxla.py

```diff
@@ -339,11 +339,15 @@ def apply_parallel_primitive(prim, *args, **params):
   # look up information in the dynamic axis env.
   dynamic_axis_env = _thread_local_state.dynamic_axis_env
   axis_name = params.pop('axis_name')
-  logical_size = lambda frame: frame.hard_size * (frame.soft_size or 1)
-  if isinstance(axis_name, (list, tuple)):
-    shape = tuple(logical_size(dynamic_axis_env[name]) for name in axis_name)
+  axis_index_groups = params.pop('axis_index_groups')
+  if axis_index_groups is not None:
+    shape = (len(axis_index_groups[0]),)
   else:
-    shape = (logical_size(dynamic_axis_env[axis_name]),)
+    logical_size = lambda frame: frame.hard_size * (frame.soft_size or 1)
+    if isinstance(axis_name, (list, tuple)):
```

Comment:
I forgot you can specify multiple axis names in a collective. The current axis_index_groups API doesn't support that, so at least comment and ideally raise an error.

Reply:
Yeah, good point. Strictly speaking it does work (indices in axis_index_groups will refer to the whole multiplexed axis formed from the multiple axis names) but I agree that we probably shouldn't do that and should just error instead.

Reply:
Oh lol wow. I mean, it's arguably the jax way to just leave this undocumented behavior as-is, in case anyone needs it... I think I would leave it but your call.

Reply:
I ended up leaving it.

Reply:
+1 to leaving it, though I'll try to check in the code to make sure I understand the semantics.